aten/src/ATen/native/native_functions.yaml

```diff
@@ -4094,7 +4094,7 @@
   dispatch:
     MkldnnCPU: mkldnn_transpose_
 
-- func: one_hot(Tensor self, int num_classes=-1) -> Tensor
+- func: one_hot(Tensor self, int num_classes=-1, ScalarType? dtype=None) -> Tensor
```

Comment:
Does `dtype` need to be optional? Maybe a default value of long would work and simplify the signatures / docs.

Reply:
A default `long` value does simplify the API. I updated the PR to support `dtype=long` as the default output dtype.